modules/playlists/src/main/java/org/opencastproject/playlists/Playlist.java

```diff
@@ -0,0 +1,240 @@
+ */
+@Entity(name = "Playlist")
+@Table(name = "oc_playlist")
+@NamedQueries({
+    @NamedQuery(
+        name = "Playlist.findById",
+        query = "SELECT p FROM Playlist p WHERE p.id = :id and p.organization = :organizationId"
+    ),
+})
+public class Playlist {
+
+  /**
+   * Workaround for generating a UUID for the id if we don't already have one.
+   *
+   * We cannot use EclipseLinks @UuidGenerator annotation for this, because it breaks our JUnit tests.
+   * We also cannot use Hibernates @GenericGenerator for this, since we lack the appropriate library (and possibly
+   * hibernate version).
+   */
+  @PrePersist
+  public void generateId() {
+    if (this.id == null) {
+      this.id = UUID.randomUUID().toString();
+    }
+  }
+
+  @Id
+  @Column(name = "id")
+  private String id;
+
+  @Column(name = "organization", nullable = false, length = 128)
+  private String organization;
+
+  @OneToMany(
+      fetch = FetchType.LAZY,
+      cascade = { CascadeType.PERSIST, CascadeType.REMOVE },
+      mappedBy = "playlist",
+      orphanRemoval = true
+  )
+  @OrderColumn(name = "position_entries")
+  private List<PlaylistEntry> entries = new ArrayList<PlaylistEntry>();
+
+  @Column(name = "title")
+  private String title;
+
+  @Column(name = "description")
+  private String description;
+
+  @Column(name = "creator")
+  private String creator;
+
+  @Column(name = "updated", nullable = false)
+  @Temporal(TemporalType.TIMESTAMP)
+  private Date updated;
+
+  @Column(name = "deletion_date")
+  @Temporal(TemporalType.TIMESTAMP)
+  protected Date deletionDate = null;
+
+  @OneToMany(
+      fetch = FetchType.LAZY,
+      cascade = { CascadeType.PERSIST, CascadeType.REMOVE },
+      mappedBy = "playlist",
+      orphanRemoval = true
+  )
+  @OrderColumn(name = "position_access_control_entries")
+  private List<PlaylistAccessControlEntry> accessControlEntries = new ArrayList<>();
+
+  /**
+   * Default constructor
+   */
+  public Playlist() {
+
+  }
+
+  public Playlist(String id, String organization, List<PlaylistEntry> entries, String title, String description,
+      String creator, Date updated, List<PlaylistAccessControlEntry> accessControlEntries) {
+    this.id = id;
+    this.organization = organization;
+    this.entries = entries;
+    this.title = title;
+    this.description = description;
+    this.creator = creator;
+    this.updated = updated;
+    this.accessControlEntries = accessControlEntries;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public String getOrganization() {
+    return organization;
+  }
+
+  public void setOrganization(String organization) {
+    this.organization = organization;
+  }
+
+  public List<PlaylistEntry> getEntries() {
+    return entries;
+  }
+
+  public void setEntries(List<PlaylistEntry> entries) {
+    for (var entry : entries) {
+      if (!entries.stream().anyMatch(e -> entry.getId() == e.getId())) {
```

Comment:
The entries are checked against themself and the statement can be simplified. 

```suggestion
      if (this.entries.stream().noneMatch(e -> entry.getId() == e.getId())) {
```